x/auth/vesting/msg_server.go

```diff
@@ -160,5 +185,180 @@ func (s msgServer) CreatePeriodicVestingAccount(goCtx context.Context, msg *type
+			}
+			return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "account %s already exists", msg.ToAddress)
+		}
+		if !ok {
+			return nil, sdkerrors.Wrapf(sdkerrors.ErrNotSupported, "account %s must be a true vesting account", msg.ToAddress)
+		}
+		if msg.FromAddress != va.FunderAddress {
+			return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "account %s can only accept grants from account %s", msg.ToAddress, va.FunderAddress)
+		}
+		newStart, newEnd, newLockupPeriods := types.DisjunctPeriods(va.StartTime, msg.GetStartTime(), va.LockupPeriods, msg.LockupPeriods)
+		newStartX, newEndX, newVestingPeriods := types.DisjunctPeriods(va.StartTime, msg.GetStartTime(),
+			va.GetVestingPeriods(), msg.GetVestingPeriods())
+		if newStart != newStartX {
+			panic("bad start time calculation")
+		}
+		va.StartTime = newStart
+		va.EndTime = max64(newEnd, newEndX)
+		va.LockupPeriods = newLockupPeriods
+		va.VestingPeriods = newVestingPeriods
+		va.OriginalVesting = va.OriginalVesting.Add(vestingCoins...)
+	} else {
+		baseAccount := ak.NewAccountWithAddress(ctx, to)
+		va = types.NewTrueVestingAccount(baseAccount.(*authtypes.BaseAccount), from, vestingCoins, msg.StartTime, msg.LockupPeriods, msg.VestingPeriods)
+		madeNewAcc = true
+	}
+
+	ak.SetAccount(ctx, va)
+
+	if madeNewAcc {
+		defer func() {
+			telemetry.IncrCounter(1, "new", "account")
+
+			for _, a := range vestingCoins {
+				if a.Amount.IsInt64() {
+					telemetry.SetGaugeWithLabels(
+						[]string{"tx", "msg", "create_true_vesting_account"},
+						float32(a.Amount.Int64()),
+						[]metrics.Label{telemetry.NewLabel("denom", a.Denom)},
+					)
+				}
+			}
+		}()
+	}
+
+	err = bk.SendCoins(ctx, from, to, vestingCoins)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+		),
+	)
+
+	return &types.MsgCreateTrueVestingAccountResponse{}, nil
+}
+
+// Clawback removes the unvested amount from a TrueVestingAccount.
+// The destination defaults to the funder address, but can be overridden.
+func (s msgServer) Clawback(goCtx context.Context, msg *types.MsgClawback) (*types.MsgClawbackResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	ak := s.AccountKeeper
+	bk := s.BankKeeper
+
+	funder, err := sdk.AccAddressFromBech32(msg.GetFunderAddress())
+	if err != nil {
+		return nil, err
+	}
+	addr, err := sdk.AccAddressFromBech32(msg.GetAddress())
+	if err != nil {
+		return nil, err
+	}
+	dest := funder
+	if msg.GetDestAddress() != "" {
+		dest, err = sdk.AccAddressFromBech32(msg.GetDestAddress())
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if bk.BlockedAddr(dest) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnauthorized, "%s is not allowed to receive funds", msg.DestAddress)
+	}
+
+	acc := ak.GetAccount(ctx, addr)
+	if acc == nil {
+		return nil, fmt.Errorf("account %s does not exist", msg.Address)
```

Comment:
use `sdkerrors` for msg server and Keeper errors

Reply:
Done, here and below.